models/commons/src/reference/reference-draft/builders.spec.ts

```diff
@@ -0,0 +1,94 @@
+import type { TReferenceDraftRest, TReferenceGraphql } from '../types';
+import {
+  ReferenceDraft,
+  ReferenceDraftGraphql,
+  ReferenceDraftRest,
+} from './index';
+
+const validateModel = (
+  model: TReferenceGraphql | TReferenceDraftRest,
+  options?: {
+    id?: string | null;
+    typeId?: string | null;
+  }
+) => {
+  expect(model).toEqual(
+    expect.objectContaining({
+      id: options?.id || expect.any(String),
+      typeId: options?.typeId || null,
+    })
+  );
+};
+
+describe('Channel model builders', () => {
+  it('builds a REST model', () => {
+    const restModel = ReferenceDraftRest.random().build();
+    validateModel(restModel);
+  });
+
+  it('builds a populated REST model', () => {
+    const restModel = ReferenceDraftRest.random()
+      .id('12345')
+      .typeId('foo')
+      .build();
+
+    validateModel(restModel, {
+      id: '12345',
+      typeId: 'foo',
+    });
+  });
+
+  it('builds a GraphQL model', () => {
+    const graphqlModel = ReferenceDraftGraphql.random().build();
+
+    validateModel(graphqlModel);
+  });
+
+  it('builds a populated GraphQL model', () => {
+    const graphqlModel = ReferenceDraftGraphql.random()
+      .id('12345')
+      .typeId('foo')
+      .build();
+
+    validateModel(graphqlModel, {
```

Comment:
Just a thought: we should also extend our Jest/Vitest matchers. So in the future we could have `expect(graphqlModel).toMatchModelContainingFields()` or a different matcher name.

Reply:
To be honest, I don't see too much value on that as consumers using that function will need to provide very much the same values that are currently used to validate a model.
I find this local helpers a little more helpful, specially in migrated models, as we have a specific set of properties to match in different local tests.

Happy to learn more about your idea in a sync if you want to.

Reply:
We can take it to our sync.